block/s3/inventory.go

```diff
@@ -125,5 +127,43 @@ func sortManifest(m *Manifest, logger logging.Logger, reader s3inventory.IReader
 			return ErrInventoryFilesRangesOverlap
 		}
 	}
+	filteredFiles := make([]inventoryFile, 0, len(m.Files))
```

Comment:
I think the filtration should happen in a seperate function that takes a slice of inventory files (i.e. `m.Files`) and returns a filtered list. This way it's much easier to unit test (this, IMO, tends to be a common case for functions that contain the word `And` :) ).